From 639e387c0fff76345734fd7fe646e140451698dd Mon Sep 17 00:00:00 2001
From: Tomas Henzl <thenzl@redhat.com>
Date: Mon, 8 Sep 2014 14:31:55 -0400
Subject: [scsi] hpsa: remove unnecessary pci_set_drvdata()

Message-id: <1410186734-19986-65-git-send-email-thenzl@redhat.com>
Patchwork-id: 92073
O-Subject: [RHEL6.6 e-stor PATCH 64/83] hpsa: remove unnecessary pci_set_drvdata()
Bugzilla: 1069185
RH-Acked-by: Maurizio Lombardi <mlombard@redhat.com>
RH-Acked-by: David Milburn <dmilburn@redhat.com>
RH-Acked-by: Ewan Milne <emilne@redhat.com>

Since commit device-core: Ensure drvdata = NULL when no driver is bound
(device-core: Ensure drvdata = NULL when no driver is bound),
the driver core clears the driver data to NULL after device_release
or on probe failure. Thus, it is not needed to manually clear the
device driver data to NULL.

Picked from 08b7e10716a518af01b07915dbb1938868bbf878
which is a treewide patch, here we change only the hpsa driver.

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/hpsa.c b/drivers/scsi/hpsa.c
index 1df75ae..4a9a07c 100644
--- a/drivers/scsi/hpsa.c
+++ b/drivers/scsi/hpsa.c
@@ -7226,7 +7226,6 @@ static void hpsa_remove_one(struct pci_dev *pdev)
  kfree(h->hba_inquiry_data);
  pci_disable_device(pdev);
  pci_release_regions(pdev);
- pci_set_drvdata(pdev, NULL);
  kfree(h);
 }
 
-- 
1.7.1